services/user-svc/internal/organization/organization.go

```diff
@@ -209,17 +203,98 @@ func (s ServiceServer) InviteMember(ctx context.Context, req *pb.InviteMemberReq
 		}
 	}
 
-	if err := db.Create(&invite).Error; err != nil {
+	// check if already an invitation exists
+	invite := Invitation{}
+
+	if err := db.Where("email = ? AND organization_id = ?", req.Email, organizationId).First(&invite).Error; err != nil {
```

Comment:
Thank you for thinking of it. Beside email and organization, we should also check if the invitation is in progress. That means, only create a new invitation if we not find one with `state == Pending` or `state == Accepted`.